services/src/main/java/org/keycloak/services/resources/account/LinkedAccountsResource.java

```diff
@@ -0,0 +1,208 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.services.resources.account;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.UUID;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.keycloak.common.util.Base64Url;
+import org.keycloak.credential.CredentialModel;
+import org.keycloak.events.Details;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.events.EventType;
+import org.keycloak.models.AccountRoles;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.FederatedIdentityModel;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.services.ErrorResponse;
+import org.keycloak.services.Urls;
+import org.keycloak.services.managers.Auth;
+import org.keycloak.services.messages.Messages;
+import org.keycloak.services.resources.Cors;
+import org.keycloak.services.validation.Validation;
+
+/**
+ * API for linking/unlinking social login accounts
+ *
+ * @author Stan Silvert
+ */
+public class LinkedAccountsResource {
+    private static final Logger logger = Logger.getLogger(LinkedAccountsResource.class);
+    
+    private final KeycloakSession session;
+    private final HttpRequest request;
+    private final ClientModel client;
+    private final EventBuilder event;
+    private final UserModel user;
+    private final RealmModel realm;
+    private final Auth auth;
+
+    public LinkedAccountsResource(KeycloakSession session, 
+                                  HttpRequest request, 
+                                  ClientModel client,
+                                  Auth auth, 
+                                  EventBuilder event, 
+                                  UserModel user) {
+        this.session = session;
+        this.request = request;
+        this.client = client;
+        this.auth = auth;
+        this.event = event;
+        this.user = user;
+        realm = session.getContext().getRealm();
+    }
+    
+    @GET
+    @Path("/")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response linkedAccounts() {
+        LinkedAccountsBean linkedAccounts = new LinkedAccountsBean(this.session, this.realm, this.user);
```

Comment:
Should have a LinkedAccountRepresentation class with plain getters/setters like all other REST endpoints do. This will allow us to create proper API docs. Also, would help to allow creating a Java account service client lib.